lib/teleterm/daemon/daemon.go

```diff
@@ -264,6 +287,15 @@ func (s *Service) FindGateway(gatewayURI string) (*gateway.Gateway, error) {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
+	gateway, err := s.findGateway(gatewayURI)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return gateway, nil
+}
+
+func (s *Service) findGateway(gatewayURI string) (*gateway.Gateway, error) {
```

Comment:
That would also prevent the duplication of gateways, which should be possible with the current implementation :thinking:

Reply:
I agree that it'd be better if it was a hashmap instead of a slice. 

Would it be sufficient for now if I leave a TODO comment next to it? In practice this being a slice shouldn't be a huge problem. Creating gateways is protected by a lock, so in practice we'd duplicate gateways only if there was a bug in the code. As far as performance is concerned, I can't imagine our users creating so many gateways that using a slice here would become a problem.